packages/interaction-ring/src/InteractionRing.tsx

```diff
@@ -0,0 +1,212 @@
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+import { css, cx } from '@leafygreen-ui/emotion';
+import { useUsingKeyboardContext } from '@leafygreen-ui/leafygreen-provider';
+import { uiColors } from '@leafygreen-ui/palette';
+
+const Mode = {
+  Light: 'light',
+  Dark: 'dark',
+} as const;
+
+type Mode = typeof Mode[keyof typeof Mode];
+
+interface ColorSet {
+  interactionRingHover: string;
+  interactionRingFocus: string;
+}
+
+const colorSets: Record<Mode, ColorSet> = {
+  [Mode.Light]: {
+    interactionRingHover: uiColors.gray.light2,
+    interactionRingFocus: '#9dd0e7',
+  },
+  [Mode.Dark]: {
+    interactionRingHover: uiColors.gray.dark1,
+    interactionRingFocus: uiColors.blue.base,
+  },
+};
+
+const baseContainerStyle = css`
+  display: inline-flex;
+  align-items: stretch;
+  position: relative;
+  z-index: 0;
+`;
+
+const baseInteractionRingStyle = css`
+  transition: all 150ms ease-in-out;
+  position: absolute;
+  z-index: -1;
+  top: 0;
+  bottom: 0;
+  left: 0;
+  right: 0;
+  pointer-events: none;
+`;
+
+function getInteractionRingStyle({
+  mode,
+  hovered,
+  focused,
+  borderRadius = '4px',
+  color,
+}: {
+  mode: Mode;
+  hovered: boolean;
+  focused: boolean;
+  borderRadius?: string;
+  color?: {
+    focused?: string;
+    hovered?: string;
+  };
+}) {
+  const hoverStyle = css`
+    box-shadow: 0 0 0 3px
+      ${color?.hovered ?? colorSets[mode].interactionRingHover};
+  `;
+
+  const focusStyle = css`
+    box-shadow: 0 0 0 3px
+      ${color?.focused ?? colorSets[mode].interactionRingFocus};
+  `;
+
+  return cx(
+    baseInteractionRingStyle,
+    css`
+      border-radius: ${borderRadius};
+    `,
+    {
+      [hoverStyle]: hovered,
+    },
+    {
+      [focusStyle]: focused,
+    },
+  );
+}
+
+const baseContentStyle = css`
+  width: 100%;
+  height: 100%;
+
+  // Form elements don't always inherit these by default
+  // https://developer.mozilla.org/en-US/docs/Learn/Forms/Styling_web_forms#Fonts_and_text
+  font-family: inherit;
+  font-size: 100%;
+`;
+
+interface InteractionRingProps {
+  darkMode?: boolean;
+  className?: string;
+  borderRadius?: string;
+  color?: {
+    focused?: string;
```

Comment:
should this be nested such that we can set colors for dark mode and light mode respectively?

Reply:
Reasonable enough, wfm.